### PR TITLE
Replace deprecated `set-output` command

### LINKED ISSUE #1937: Replace deprecated `set-output` command
## Problem
The CI logs are full of warnings about the outdated `set-output` command.

## Solution
GitHub actions [introduced](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/) a new set of environment files to manage state and output.

In the `ghcr` workflow, 
https://github.com/WordPress/openverse-frontend/blob/952b8e830aff7437584cdc23dab469e859739b2c/.github/workflows/ghcr.yml#L73

`print(f"::set-output name=version::{version}")`
should be replaced with 
`print(f"{version}={version}") >> $GITHUB_OUTPUT`

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1937 by @obulat

## Description

This PR replaces the outdated `set-output` command with the recommended value: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands

## Testing Instructions

The CI should work

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .github/workflows/ghcr.yml [hunk 1]
```diff
@@ -62,15 +62,16 @@ jobs:
...
           metadata = json.loads('${{ steps.metadata.outputs.json }}')
           version = metadata["labels"]["org.opencontainers.image.version"]
           if version == "main":
               # get first version that isn't main
               for tag in metadata["tags"]:
                   if ":main" not in tag:
                       version = tag.split(":")[1]
-
-          print(f"::set-output name=version::{version}")
+          with open(os.environ.get("GITHUB_STATE"), "a") as state:
```
Comment (krysal) [inline .github/workflows/ghcr.yml:73]:
```suggestion
          with open(os.environ.get("GITHUB_OUTPUT"), "a") as state:
```
Changing to `GITHUB_OUTPUT` should keep the rest of the flow the same. Curious if there is an advantage on using `GITHUB_STATE`?

Comment (obulat) [reply .github/workflows/ghcr.yml:73]: You're right, @krysal, I mixed up the examples with `state` and `output` here: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/#examples

**Review by zackkrida — APPROVED** [review]
LGTM!

**Review by krysal — APPROVED** [review]
LGTM! The failing tests seems to be an unrelated thing.

**Pull request merged**